### PR TITLE
Matching V2

### PR DESCRIPTION
**Pull request by jcarrete5**
image

### CODE REVIEW

**Review by AyalaCnaan — COMMENTED** [review]
As far as I can tell this is good. I am confused by the screen shot, why is email on there?

**Comment by jcarrete5** [comment]
> As far as I can tell this is good. I am confused by the screen shot, why is email on there?

The UI is very much subject to change. As of right now, the match card doesn't respect user privacy so the email shows up. I have noted this as a bug in [clickup](https://app.clickup.com/t/t95jvw).

File: frontend/src/FullPageRoutes/Matching.tsx [hunk 1]
```diff
@@ -6,11 +6,59 @@ import {
...
+  last_name: string
+  gender: string
+  birth_date: string
+  email: string
+  phone_number: string
+  interests: Interest[]
+  has_intro: boolean
+}
+
+type InterestID = number
```
Comment (mneedleman) [inline frontend/src/FullPageRoutes/Matching.tsx:40]: Why is this necessary?

File: frontend/src/FullPageRoutes/Matching.tsx [hunk 2]
```diff
@@ -6,11 +6,59 @@ import {
...
+  OTHER = "O",
+}
+
+class FilterState {
+  static readonly ageLimits = { min: 18, max: 130 }
+  readonly minAge = FilterState.ageLimits.min
+  readonly maxAge = FilterState.ageLimits.max
+  readonly genders = Object.values(Gender)
+  readonly interests: Interest[] = []
+}
```
Comment (mneedleman) [inline frontend/src/FullPageRoutes/Matching.tsx:60]: I'm pretty hesitant to take an OOP approach, but I can't deny that's concise and clean. However, as a 131 year old, I feel discriminated against

File: frontend/src/utils/fetch.tsx [hunk 3]
```diff
@@ -65,3 +65,36 @@ export async function request<T>({
...
+        } else {
+          throw new Error(`Failed to encode key/value pair: ${key}=${val}`)
+        }
+        break
+      default:
+        throw new Error(`Failed to encode key/value pair: ${key}=${val}`)
+    }
+  }
+  return "?" + params.join("&")
+}
```
Comment (mneedleman) [inline frontend/src/utils/fetch.tsx:100]: If we had more time, I'd say we should explore how RamdaJS might help clean this up. But looks good

**Review by mneedleman — APPROVED** [review]

File: frontend/src/FullPageRoutes/Matching.tsx (hunk 1 above)
Comment (jcarrete5) [reply frontend/src/FullPageRoutes/Matching.tsx:40]: It's not. I was trying to avoid using the `Interest` type in the filter state so it would be easier to encode in the URL. I don't need it anymore but I didn't feel it was worth removing

Comment (mneedleman) [reply frontend/src/FullPageRoutes/Matching.tsx:40]: Fair

**Review by AyalaCnaan — APPROVED** [review]
